portfolio/src/main/webapp/script.js

```diff
@@ -12,39 +12,70 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** @param slideIndex slide currently being displayed */
 var slideIndex = 1;
+
+/**
+ * Supported class states for dots and slides.
+ * @enum {string}
+ */
+const states = {
+  SHOW_DOT: ' active',
+  HIDE_DOT: '',
+  SHOW_SLIDE: 'block',
+  HIDE_SLIDE: 'none',
+}
+
 showSlides(slideIndex);
 
-// Next and previous controls
+/**
+ * Function computes which slide should be shown and class showSlides on that
```

Comment:
Nit: this comment can also cause confusion because the user does not know what showSlides does. Instead of referring to other methods just explain what the method does. You could write:

Computes which slide should be show and then shows that specific slide.  Invoked when either arrows are pressed.
@param ...

Reply:
Done